### PR TITLE
made TransformDirection enum string based; added _missing_ method to raise error when value missing

### PR DESCRIPTION
**Pull request by snowman2**
Addresses #298 and https://github.com/pyproj4/pyproj/pull/288#issuecomment-490122764

Uses idea from https://github.com/pyproj4/pyproj/pull/288#issuecomment-490123157 to better handle invalid input.

@jorisvandenbossche

### CODE REVIEW

**Comment by snowman2** [comment]
Looks like `_missing_` only works post Python 3.5. Shame, it is really nice.

**Comment by snowman2** [comment]
@jswhit, that sounds good. I will go ahead and merge. Would you be okay with me making a release as well (minus uploading to pypi)?

**Comment by snowman2** [comment]
I was thinking about making a release along with PROJ 6.1.0 release on May 15. Is that okay? Or would you prefer to be able to review everything before then (which is fine - no rush on my end)?

**Comment by snowman2** [comment]
> I was thinking about making a release along with PROJ 6.1.0 release on May 15. Is that okay?

I think that does not make much sense. I think waiting until you are ready would be the better decision in this case. Better to have a good review before the release since there isn't a rush. I was just getting ahead of myself there.

File: pyproj/enums.py [hunk 1]
```diff
@@ -35,14 +35,14 @@ class ProjVersion(IntEnum):
...
 
-class TransformDirection(IntEnum):
+class TransformDirection(Enum):
     """
     Supported transform directions
     """
 
     #: Forward direction
-    FORWARD = 1
+    FORWARD = "FORWARD"
```
Comment (jorisvandenbossche) [inline pyproj/enums.py]: Small remark: we could also opt for lower case? (that might be a little bit nicer in user code (`transform(..., direction='inverse')` vs `transform(..., direction='INVERSE')`)

Just a remark, no strong preference either way!

Comment (snowman2) [reply pyproj/enums.py]: I thought about doing that, and decided that the upper case version is consistent with the upper case WKT enum strings.

Comment (jorisvandenbossche) [reply pyproj/enums.py]: Yep, on the other hand, WKT is an existing abbreviation that is written in upper case, while forward/inverse are normal words.

Comment (snowman2) [reply pyproj/enums.py]: That's true. What would be nice is a case insensitive solution.

Comment (snowman2) [reply pyproj/enums.py]: That would preserve past behavior as that's how I had it before.

**Pull request merged**